src/schema/feeds.ts

```diff
@@ -686,6 +688,65 @@ const applyFeedPaging = (
   return newBuilder;
 };
 
+const feedPageGeneratorWithPin: PageGenerator<
+  GQLPost,
+  FeedArgs,
+  FeedPage,
+  unknown
+> = {
+  connArgsToPage: ({ first, after }: FeedArgs) => {
+    const cursor = getCursorFromAfter(after);
+    const limit = Math.min(first || 30, 50) + 1;
+
+    return cursor
+      ? { limit, timestamp: new Date(parseInt(cursor)) }
+      : { limit };
+  },
+  nodeToCursor: (_, __, node) => {
+    const createdAt = new Date(node.createdAt);
+    const timestamp =
+      node.type === PostType.Welcome
+        ? changeYearToNextYear(createdAt)
+        : createdAt;
+
+    return base64(`time:${new Date(timestamp).getTime()}`);
+  },
+  hasNextPage: (page, nodesSize) => page.limit === nodesSize,
+  hasPreviousPage: (page) => !!page.timestamp,
+  transformNodes: (page, nodes) => nodes.slice(0, page.limit - 1),
+};
+
+const applyFeedPagingWithPin = (
+  ctx: Context,
+  page: FeedPage,
+  builder: SelectQueryBuilder<Post>,
+  alias,
+): SelectQueryBuilder<Post> => {
+  const priority = `CASE WHEN ${alias}.type = '${PostType.Welcome}' THEN 1 ELSE 0 END`;
+  let newBuilder = builder
+    .addSelect(priority, 'priority')
+    .limit(page.limit)
+    .orderBy(`priority`, 'DESC')
+    .addOrderBy(`${alias}."createdAt"`, 'DESC');
+
+  if (page.timestamp) {
+    // to ensure pinned post (or welcome post) be pinned on top of the feed, we need to ensure they have greater date than the rest of the result
+    // that is solved by making the year of pinned posts to the next year and since we retained the other parts of the date, sorting should be accurate.
+    // the current condition is about being a welcome post, but on v6 this will be based on the `pinned` property.
```

Comment:
So the next year hack is just temporary for v5?

Reply:
Yes, this will use the pinned property for the condition once we introduce it in v6.

Reply:
Cool, tnx for clarification.

Reply:
Though we will keep the next year mutation until we find a better solution for pagination including two properties.